wandb/apis/public.py

```diff
@@ -1916,22 +1916,27 @@ def file(self, name):
     def upload_file(self, path, root="."):
         """
         Arguments:
-            path (str): name of file to upload.
-            root (str): the root path to save the file relative to.  i.e.
-                If you want to have the file saved in the run as "my_dir/file.txt"
-                and you're currently in "my_dir" you would set root to "../"
+            path (str): local path to the file to upload.
+            root (str): the root path to save the file in wandb.
 
         Returns:
             A `File` matching the name argument.
         """
+        path = os.path.normpath(os.path.join(os.getcwd(), os.path.relpath(path, os.getcwd())))
+        if not os.path.isfile(path):
```

Comment:
Should we check not just this is a file but also that it has an extension? Or is that already implied in `os.path.isfile`?

Reply:
```suggestion
        if not os.path.isfile(path) and os.path.splitext(path)[1] != "":
```

Reply:
Suggestion above is the suggested logic, not the logic to implement actually, as we still need to check whether `os.path.isfile` checks whether it contains an extension or not!

Reply:
`os.path.isfile(path)` just checks whether it's a file or not, but it does not take into consideration the extension, as a file does not need to have an extension. So we'll need to check also that it contains an extension, as `wandb` doesn't support uploading files without extension. Indeed if there's a file named `test` and a directory named `test`, the UI won't let you see the `test` file, it'll just redirect you to the `test` directory.